src/super_gradients/examples/predict/detection_predict_with_labels.py

```diff
@@ -0,0 +1,26 @@
+from super_gradients.common.object_names import Models
+from super_gradients.training import models
+from pathlib import Path
+
+# Note that currently only YoloX, PPYoloE and YOLO-NAS are supported.
+from super_gradients.training.datasets import COCODetectionDataset
+
+model = models.get(Models.YOLO_NAS_L, pretrained_weights="coco")
+mini_coco_data_dir = str(Path(__file__).parent.parent.parent.parent.parent / "tests" / "data" / "tinycoco")
+
+dataset = COCODetectionDataset(
+    data_dir=mini_coco_data_dir, subdir="images/val2017", json_file="instances_val2017.json", input_dim=None, transforms=[], cache_annotations=False
+)
+
+# x's are np.ndarrays images of shape (H,W,3)
+# y's are np.ndarrays of shape (num_boxes,x1,y1,x2,y2,class_id)
+x1, y1, _ = dataset[0]
+x2, y2, _ = dataset[1]
+
+# images from COCODetectionDataset are RGB and images as np.ndarrays are expected to be BGR
+x2 = x2[:, :, ::-1]
+x1 = x1[:, :, ::-1]
+
+predictions = model.predict([x1, x2], target_bboxes=[y1[:, :4], y2[:, :4]], target_class_ids=[y1[:, 4], y2[:, 4]], target_bboxes_format="xyxy")
+predictions.show()
```

Comment:
Why do we add these params to `predict` instead of `show` ? Eventually, it's just a visualization adding, and does not affect the predictions.
One practical downside of adding to predict is that this has to propagate to the DetectionPipeline, and it adds complexity to many classes/functions. Another thing is that if we run a prediction, it would make sense to have option to show or not show the GT without rerunning the prediction.

Would it be complex to move the logic to the `predictions.show()` function? From what it looks, I think the code would actually me much simpler, but maybe I'm wrong

Reply:
There is a downside to that - saving images will not consider the targets if we go that way.
I argue that from some threshold of the image count, users will tend to use save rather then show().
Also, sometimes using show() is not an option.

Reply:
Overall the PR looks great to me.

I agree with @Louis-Dupont - to me the advantage of putting `targets` to predict method signature is ill-justified decision. It violates SRP, makes signature of the method more complicated, I frankly I see no real advantage of solving it this way.

To address your counter-arguments:

> There is a downside to that - saving images will not consider the targets if we go that way.
True, unless you add `targets` to `save(...)` method.

In conclusion, I feel that having `def save(..., target_boxes, target_labels` and `def show(..., target_boxes, target_labels)` would be much cleaner API design.

Under-the-hood implementation looks already LGTM, my only concert is user-facing API.

Reply:
Alright you guys win (:
de413502d01b617481b509f27da06fe092ec1fa1